File: code/ch02/2.6.1.maxSubArray.java
import java.lang.Math;

public class Solution {
  public int maxSubArray(int[] nums) {
    int n = nums.length;
    int maxSum = -0x7fffffff, total = 0;
    for (int i = 0; i < n; i++) {
      total = 0;
      for (int j = i; j < n; j++) {
        total += nums[j];
        maxSum = Math.max(maxSum, total);
      }
    }
    return maxSum;
  }
}

File: code/ch02/2.6.1.maxSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        maxSum = float("-inf")
        total = 0
        for i in range(n):
            total = 0
            for j in range(i, n):
                total += nums[j]
                maxSum = max(maxSum, total)

        return maxSum
